enterprise/server/invocation_stat_service/invocation_stat_service.go

```diff
@@ -135,79 +140,93 @@ func flattenTrendsQuery(innerQuery string) string {
 
-	if user := req.GetQuery().GetUser(); user != "" {
+	if user := tq.GetUser(); user != "" {
 		q.AddWhereClause("user = ?", user)
 	}
 
-	if host := req.GetQuery().GetHost(); host != "" {
+	if host := tq.GetHost(); host != "" {
 		q.AddWhereClause("host = ?", host)
 	}
 
-	if repoURL := req.GetQuery().GetRepoUrl(); repoURL != "" {
+	if repoURL := tq.GetRepoUrl(); repoURL != "" {
 		q.AddWhereClause("repo_url = ?", repoURL)
 	}
 
-	if branchName := req.GetQuery().GetBranchName(); branchName != "" {
+	if branchName := tq.GetBranchName(); branchName != "" {
 		q.AddWhereClause("branch_name = ?", branchName)
 	}
 
-	if command := req.GetQuery().GetCommand(); command != "" {
+	if command := tq.GetCommand(); command != "" {
 		q.AddWhereClause("command = ?", command)
 	}
 
-	if commitSHA := req.GetQuery().GetCommitSha(); commitSHA != "" {
+	if commitSHA := tq.GetCommitSha(); commitSHA != "" {
 		q.AddWhereClause("commit_sha = ?", commitSHA)
 	}
 
 	roleClauses := query_builder.OrClauses{}
-	for _, role := range req.GetQuery().GetRole() {
+	for _, role := range tq.GetRole() {
 		roleClauses.AddOr("role = ?", role)
 	}
 	if roleQuery, roleArgs := roleClauses.Build(); roleQuery != "" {
 		q.AddWhereClause("("+roleQuery+")", roleArgs...)
 	}
 
-	if start := req.GetQuery().GetUpdatedAfter(); start.IsValid() {
+	if start := tq.GetUpdatedAfter(); start.IsValid() {
 		q.AddWhereClause("updated_at_usec >= ?", start.AsTime().UnixMicro())
 	} else {
 		// If no start time specified, respect the lookback window field if set,
 		// or default to 7 days.
 		// TODO(bduffany): Delete this once clients no longer need it.
 		lookbackWindowDays := 7 * 24 * time.Hour
-		if w := req.GetLookbackWindowDays(); w != 0 {
-			if w < 1 || w > 365 {
+		if lookback != 0 {
+			if lookback < 1 || lookback > 365 {
 				return status.InvalidArgumentErrorf("lookback_window_days must be between 0 and 366")
 			}
-			lookbackWindowDays = time.Duration(w*24) * time.Hour
+			lookbackWindowDays = time.Duration(lookback*24) * time.Hour
 		}
 		q.AddWhereClause("updated_at_usec >= ?", time.Now().Add(-lookbackWindowDays).UnixMicro())
 	}
 
-	if end := req.GetQuery().GetUpdatedBefore(); end.IsValid() {
+	if end := tq.GetUpdatedBefore(); end.IsValid() {
 		q.AddWhereClause("updated_at_usec < ?", end.AsTime().UnixMicro())
 	}
 
-	statusClauses := toStatusClauses(req.GetQuery().GetStatus())
+	statusClauses := toStatusClauses(tq.GetStatus())
 	statusQuery, statusArgs := statusClauses.Build()
 	if statusQuery != "" {
 		q.AddWhereClause(fmt.Sprintf("(%s)", statusQuery), statusArgs...)
 	}
 
-	q.AddWhereClause(`group_id = ?`, groupID)
-	q.SetGroupBy("name")
+	if tq.GetMinimumDuration().GetSeconds() != 0 {
+		q.AddWhereClause(`duration_usec >= ?`, tq.GetMinimumDuration().GetSeconds()*1e6)
+	}
+	if tq.GetMaximumDuration().GetSeconds() != 0 {
+		q.AddWhereClause(`duration_usec <= ?`, tq.GetMaximumDuration().GetSeconds()*1e6)
+	}
```

Comment:
We've screwed up usec/nsec/sec before. You could do:

tq.GetMaximumDuration().AsDuration().Microseconds()

Reply:
Sure yeah yes done